control-plane/stor-port/src/types/v0/transport/nexus.rs

```diff
@@ -79,6 +79,88 @@ impl From<Nexus> for models::Nexus {
 
 rpc_impl_string_uuid!(NexusId, "UUID of a nexus");
 
+/// A gRPC request type for creating snapshot of a nexus, which essentially
+/// means a snapshot of all(or selected) replicas associated with that nexus.
+pub struct CreateNexusSnapshot {
+    /// uuid of the nexus.
+    pub nexus: NexusId,
+    /// uuid of the entity, nexus in this case.
+    pub entity: uuid::Uuid,
+    /// A transaction id for this request.
+    pub txn_id: String,
+    /// name of the snapshot to be created.
+    pub name: String,
```

Comment:
Should we use a `SnapshotId` type and get rid of the name? cc @tiagolobocastro

Reply:
Additionally, I'm not clear why nexus id and entity both are required by the interface here. @mtzaurus

Reply:
`Should we use a SnapshotId type and get rid of the name? cc @tiagolobocastro`
I was chatting with Tiago yesterday, we'll have a round of meeting to revisit few of this fields.

Reply:
Sure.